shell/browser/native_window_mac.mm

```diff
@@ -520,6 +520,9 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
 }
 
 void NativeWindowMac::RepositionTrafficLights() {
+  // Ensure maximizable options retain pre-existing state.
+  [[window_ standardWindowButton:NSWindowZoomButton] setEnabled:maximizable_];
+
```

Comment:
:+1: on handling this on theme changes, but putting this code in `RepositionTrafficLights()` is coincidental cohesion -- for example, `NativeWindowMac::SetTitle()` now refreshes the maximizable button for no reason.

Maybe the browser ui thread tasks posted in `NativeWindowMac::OnNativeThemeUpdated()` should be a new `OnNativeThemeUpdatedInUI()` type function that calls both the maximizable update and also `RepositionTrafficLights()`

Reply:
I've fixed the cohesion and made it such that `SetMaximizable` is a no-op if there is no state change. Unfortunately, moving the update call into OnNativeThemeUpdated doesn't fix the bug and continues to show the bad behavior, so this seemed like the most effective solution which wouldn't require a bit of a refactor which seemed scope-creepy.

Reply:
I wasn't suggesting moving the upate call to OnNativeThemeUpdated -- I was suggesting that OnNativeThemeUpdated to call a new wrapper function OnNatveThemeUpdatedOnUI and _that_ function call both SetMaximizable and RepositionTrafficLights.

Similar idea here wrt having OnNativeThemeUpdated be a pass-through to a sibling OnUI function: https://github.com/electron/electron/blob/master/shell/browser/api/electron_api_native_theme.cc#L38

Reply:
@ckerr this gets called whenever macOS redraws a window - it's fine that it does both of these things i'd say although i've renamed it to `RedrawTrafficLights` to clarify intent. macOS/Cocoa also optimizes this away so it's not called excessively when not needed.